src/os/flatpak.zig

```diff
@@ -501,3 +501,61 @@ pub const FlatpakHostCommand = struct {
 
     fn noopCallback(_: ?*anyopaque, _: *xev.Loop, _: *Completion, _: WaitError!u8) void {}
 };
+
+/// A struct to help retrive information about the running Flatpak.
+///
+/// Requires GLib to be available and linked.
+pub const FlatpakInfo = struct {
+    const Self = @This();
+    const c = @cImport({
+        @cInclude("glib.h");
+    });
+
```

Comment:
New code that uses the raw C GTK interfaces is a non-starter, _expecially_ outside `src/gtk`. If you're not sure how to implement the interface that Mitchell described let us know and we can help.

Reply:
CI never got a chance to try macOS builds because of other errors, but this is not going to compile on macOS.

Reply:
> If you're not sure how to implement the interface that Mitchell described let us know and we can help.

I'd love to have some guidance on this.

What has been bugging me is that Flatpak comes with the distinction of host vs. app paths and it is not clear to me how to model this well.

Right now my solution is basically just more logic into termio, which isn't really great.